NEW BRANCH: v2: Update Provider to use Ignition v2.x / Ignition Config Spec v3.x

LorbusChris opened the pull request:
:warning: This PR should not be merged! :warning:

Instead, a v2 branch should be created for these changes so that this version can be distributed alongside v1.

So far this repository has been containing a provider for Ignition 0.x / spec 2.x.

The current release stream of Ignition is 2.x. This branch contains a provider for it, supporting Ignition config spec 3.x.

LorbusChris committed:
v2: Update Provider to use Ignition v2.1.1

Ignition v2 uses Ignition Config Spec v3.0.0

LorbusChris committed:
Update vendor

LorbusChris committed:
Add overwrite arg to files, dirs and links

LorbusChris commented:
Please reach out to me if you're interested in co-maintaining this branch over at https://github.com/community-terraform-providers/terraform-provider-ignition

LorbusChris committed:
Update vendor

Update all dependencies with `go get -u`

LorbusChris committed:
Update provider to use Ignition Spec v3.1

LorbusChris committed:
Fixups for errcheck

LorbusChris committed:
Merge pull request #1 from LorbusChris/spec31

Update provider to use Ignition Spec v3.1

statsperformjanos commented:
Thanks @LorbusChris! To upgrade to 3.0.0 is really necessary to support Fedora CoreOS: https://github.com/coreos/fedora-coreos-tracker/issues/31

aries1980 commented:
Thanks @LorbusChris! To upgrade to 3.0.0 is really necessary to support Fedora CoreOS: https://github.com/coreos/fedora-coreos-tracker/issues/31

yussufsh committed:
Add build release script

Signed-off-by: Yussuf Shaikh <yussuf.shaikh@ibm.com>

LorbusChris committed:
Merge pull request #2 from yussufsh/build

Add build release script

yussufsh committed:
Add workflow and releaser for registry setup (#3)

egeturgay committed:
update doc to comply with HCLv2 and Ignition v2 (#5)

LorbusChris committed:
Replace docs for `append` with `merge`

The `append` field is not supported in Ignition config spec v3 anymore.

Fixes  #7

LorbusChris committed:
Remove Changelog

LorbusChris committed:
Update module path

Rename from `github.com/terraform-providers/terraform-provider-ignition/v2`
to `github.com/community-terraform-providers/terraform-provider-ignition/v2`

bgilbert committed:
Merge pull request #11 from LorbusChris/v2-cleanup

V2 cleanup

nccurry committed:
Issue 17: Docs don't match provider implementation (#18)

* Update disk.html.md to match resource_ignition_disk.go.
* Update filesystem.html.md to match resource_ignition_filesystem.go.
* Update link.html.md to match resource_ignition_link.go.
* Remove documentation for networkd_unit.html.md without matching resource.
* Update disk.html.md to match resource_ignition_disk.go.
* Update mib to MiB for clarity.
* Update path description to match ignition spec documentation.

Co-authored-by: ncc <ncc@nickcurry.com>

yselkowitz committed:
Re-vendor golang.org/x/sys (#19)

This is needed to build for MacOS with Go 1.18.

yussufsh committed:
Add support for darwin_arm64

Fixes #21

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

bgilbert committed:
gofmt for Go 1.18 (#24)

yussufsh committed:
Update go version

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update vendor

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

mathlu committed:
Add kernelArguments

Updates ignition vendor to v2.11.0  and schema version to v3.3.

Adds support for KernelArguments.

Fixes #20

yussufsh committed:
go lint and build gates

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
fix lint errors

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
update the ignition/config spec versions in docs

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
doc updates for kernel arguments

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
cleanup go vendor dir

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
add support for httpHeaders

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
update doc for http_headers

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump golang.org/x/text from 0.3.3 to 0.3.8

Bumps [golang.org/x/text](https://github.com/golang/text) from 0.3.3 to 0.3.8.
- [Release notes](https://github.com/golang/text/releases)
- [Commits](https://github.com/golang/text/compare/v0.3.3...v0.3.8)

---
updated-dependencies:
- dependency-name: golang.org/x/text
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go from 1.30.28 to 1.34.0

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.30.28 to 1.34.0.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/v1.34.0/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.30.28...v1.34.0)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golang.org/x/crypto from 0.0.0-20200429183012-4b2356b1ed79 to 0.1.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.0.0-20200429183012-4b2356b1ed79 to 0.1.0.
- [Commits](https://github.com/golang/crypto/commits/v0.1.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.11.0 to 2.14.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.11.0 to 2.14.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.11.0...v2.14.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.0.0-20200602114024-627f9648deb9 to 0.7.0

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.0.0-20200602114024-627f9648deb9 to 0.7.0.
- [Commits](https://github.com/golang/net/commits/v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/ulikunitz/xz from 0.5.7 to 0.5.8

Bumps [github.com/ulikunitz/xz](https://github.com/ulikunitz/xz) from 0.5.7 to 0.5.8.
- [Commits](https://github.com/ulikunitz/xz/compare/v0.5.7...v0.5.8)

---
updated-dependencies:
- dependency-name: github.com/ulikunitz/xz
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Create dependabot.yml

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/kisielk/errcheck from 1.2.0 to 1.6.3

Bumps [github.com/kisielk/errcheck](https://github.com/kisielk/errcheck) from 1.2.0 to 1.6.3.
- [Release notes](https://github.com/kisielk/errcheck/releases)
- [Commits](https://github.com/kisielk/errcheck/compare/v1.2.0...v1.6.3)

---
updated-dependencies:
- dependency-name: github.com/kisielk/errcheck
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk from 1.12.0 to 1.17.2

Bumps [github.com/hashicorp/terraform-plugin-sdk](https://github.com/hashicorp/terraform-plugin-sdk) from 1.12.0 to 1.17.2.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/v1.17.2/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v1.12.0...v1.17.2)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/go-getter

Bumps [github.com/hashicorp/go-getter](https://github.com/hashicorp/go-getter) from 1.4.2-0.20200106182914-9813cbd4eb02 to 1.7.0.
- [Release notes](https://github.com/hashicorp/go-getter/releases)
- [Changelog](https://github.com/hashicorp/go-getter/blob/main/.goreleaser.yml)
- [Commits](https://github.com/hashicorp/go-getter/commits/v1.7.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/go-getter
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/Masterminds/goutils from 1.1.0 to 1.1.1

Bumps [github.com/Masterminds/goutils](https://github.com/Masterminds/goutils) from 1.1.0 to 1.1.1.
- [Release notes](https://github.com/Masterminds/goutils/releases)
- [Changelog](https://github.com/Masterminds/goutils/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Masterminds/goutils/compare/v1.1.0...v1.1.1)

---
updated-dependencies:
- dependency-name: github.com/Masterminds/goutils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Update index page with version details

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.14.0 to 2.16.2

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.14.0 to 2.16.2.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.14.0...v2.16.2)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Bump github.com/coreos/vcontext from v0.0.0-20230201181013-d72178a18687 to v0.0.0-20230530201827-5a3e46ba99f8

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Upgrade ignition spec to v3.4

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add missing compression in ignition config

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update docs for verification description

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
fix error messages for length checks in tests

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
ignition_file: add support for http_headers

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add support for security tls CAs

Fixes #8

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update index page with version details

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Replace deprecated goreleaser option --rm-dist with --clean

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Allow dependabot to upgrade github-actions

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-go from 3 to 4

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 3 to 4.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Upgrade go v1.20

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add timeout for golangci-lint action

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump goreleaser/goreleaser-action from 4 to 5

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 4 to 5.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump crazy-max/ghaction-import-gpg from 5 to 6

Bumps [crazy-max/ghaction-import-gpg](https://github.com/crazy-max/ghaction-import-gpg) from 5 to 6.
- [Release notes](https://github.com/crazy-max/ghaction-import-gpg/releases)
- [Commits](https://github.com/crazy-max/ghaction-import-gpg/compare/v5...v6)

---
updated-dependencies:
- dependency-name: crazy-max/ghaction-import-gpg
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.12.0 to 0.17.0

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.12.0 to 0.17.0.
- [Commits](https://github.com/golang/net/compare/v0.12.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump google.golang.org/grpc from 1.56.1 to 1.56.3

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.56.1 to 1.56.3.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.56.1...v1.56.3)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

n0rad committed:
Add support for luks resource (#63)

* up

* up

* up

* up

* Update website/docs/d/luks.html.md

Co-authored-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

* Update website/docs/d/luks.html.md

Co-authored-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

* up

* up

* up

* up

* up

* Update website/docs/d/luks.html.md

Co-authored-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

* Update website/docs/d/luks.html.md

Co-authored-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

---------

Co-authored-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.16.2 to 2.17.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.16.2 to 2.17.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.16.2...v2.17.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Update Terraform plugin SDK v2

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Fix variable type in tests

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump golang.org/x/crypto from 0.15.0 to 0.17.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.15.0 to 0.17.0.
- [Commits](https://github.com/golang/crypto/compare/v0.15.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.30.0 to 2.31.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.30.0 to 2.31.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.30.0...v2.31.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Update version details for ignition 2.17.0

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump actions/setup-go from 4 to 5

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 4 to 5.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/cloudflare/circl from 1.3.3 to 1.3.7

Bumps [github.com/cloudflare/circl](https://github.com/cloudflare/circl) from 1.3.3 to 1.3.7.
- [Release notes](https://github.com/cloudflare/circl/releases)
- [Commits](https://github.com/cloudflare/circl/compare/v1.3.3...v1.3.7)

---
updated-dependencies:
- dependency-name: github.com/cloudflare/circl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/kisielk/errcheck from 1.6.3 to 1.7.0

Bumps [github.com/kisielk/errcheck](https://github.com/kisielk/errcheck) from 1.6.3 to 1.7.0.
- [Release notes](https://github.com/kisielk/errcheck/releases)
- [Commits](https://github.com/kisielk/errcheck/compare/v1.6.3...v1.7.0)

---
updated-dependencies:
- dependency-name: github.com/kisielk/errcheck
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.31.0 to 2.32.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.31.0 to 2.32.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.31.0...v2.32.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.32.0 to 2.33.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.32.0 to 2.33.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.32.0...v2.33.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golangci/golangci-lint-action from 3 to 4

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 3 to 4.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.17.0 to 2.18.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.17.0 to 2.18.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.17.0...v2.18.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump google.golang.org/protobuf from 1.32.0 to 1.33.0

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.21.0 to 0.23.0 in the go_modules group

Bumps the go_modules group with 1 update: [golang.org/x/net](https://github.com/golang/net).


Updates `golang.org/x/net` from 0.21.0 to 0.23.0
- [Commits](https://github.com/golang/net/compare/v0.21.0...v0.23.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golangci/golangci-lint-action from 4 to 5

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 4 to 5.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

X-Cli committed:
Add missing mount_options parameter in filesystem

Closes #79

There was some mixup between the "mount" option from Ignition v2 and
the "options" and "mountOptions" from Ignition v3.1+

This commit fixes the names of the related functions, adds mountOptions
support and updates the documentation accordingly.

Signed-off-by: Florian Maury <florian.maury@metempsychose.fr>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.33.0 to 2.34.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.33.0 to 2.34.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.33.0...v2.34.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golangci/golangci-lint-action from 5 to 6

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 5 to 6.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v5...v6)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump goreleaser/goreleaser-action from 5 to 6

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 5 to 6.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v5...v6)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the go_modules group with 2 updates

Bumps the go_modules group with 2 updates: [golang.org/x/crypto](https://github.com/golang/crypto) and [golang.org/x/net](https://github.com/golang/net).


Updates `golang.org/x/crypto` from 0.23.0 to 0.31.0
- [Commits](https://github.com/golang/crypto/compare/v0.23.0...v0.31.0)

Updates `golang.org/x/net` from 0.23.0 to 0.25.0
- [Commits](https://github.com/golang/net/compare/v0.23.0...v0.25.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
  dependency-group: go_modules
- dependency-name: golang.org/x/net
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.18.0 to 2.20.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.18.0 to 2.20.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.18.0...v2.20.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Upgrade go version

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update index for include 2.18.0 ignition version

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update index for include 2.20.0 ignition version

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.30.0 to 0.33.0 in the go_modules group

Bumps the go_modules group with 1 update: [golang.org/x/net](https://github.com/golang/net).


Updates `golang.org/x/net` from 0.30.0 to 0.33.0
- [Commits](https://github.com/golang/net/compare/v0.30.0...v0.33.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Fix go releaser

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add terraform-registry-manifest file

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add missing partitions specs

Fixes #85

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/kisielk/errcheck from 1.7.0 to 1.9.0

Bumps [github.com/kisielk/errcheck](https://github.com/kisielk/errcheck) from 1.7.0 to 1.9.0.
- [Release notes](https://github.com/kisielk/errcheck/releases)
- [Commits](https://github.com/kisielk/errcheck/compare/v1.7.0...v1.9.0)

---
updated-dependencies:
- dependency-name: github.com/kisielk/errcheck
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Update support for 3.5 ignition spec

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Add raid options

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Correct implementation for files.contents

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.35.0 to 0.36.0 in the go_modules group

Bumps the go_modules group with 1 update: [golang.org/x/net](https://github.com/golang/net).


Updates `golang.org/x/net` from 0.35.0 to 0.36.0
- [Commits](https://github.com/golang/net/compare/v0.35.0...v0.36.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.36.0 to 0.38.0 in the go_modules group

Bumps the go_modules group with 1 update: [golang.org/x/net](https://github.com/golang/net).


Updates `golang.org/x/net` from 0.36.0 to 0.38.0
- [Commits](https://github.com/golang/net/compare/v0.36.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

fhemberger committed:
Allow using user/group instead of uid/gid for directory, file and link

dependabot[bot] committed:
Bump actions/setup-go from 5 to 6

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 5 to 6.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 4 to 5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Fix for linter issue

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.34.0 to 2.38.1

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.34.0 to 2.38.1.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.34.0...v2.38.1)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-version: 2.38.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.20.0 to 2.25.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.20.0 to 2.25.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.20.0...v2.25.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-version: 2.25.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
Fix linter issues QF1001 and QF1008

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump golangci/golangci-lint-action from 6 to 8

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 6 to 8.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v6...v8)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-version: '8'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
fix code scan alert: Workflow does not contain permissions

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
update the ignition version in docs

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.25.0 to 2.25.1

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.25.0 to 2.25.1.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.25.0...v2.25.1)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-version: 2.25.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 5 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 5 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.38.1 to 2.38.2

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.38.1 to 2.38.2.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.38.1...v2.38.2)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-version: 2.38.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/cloudflare/circl

Bumps the go_modules group with 1 update in the / directory: [github.com/cloudflare/circl](https://github.com/cloudflare/circl).


Updates `github.com/cloudflare/circl` from 1.6.1 to 1.6.3
- [Release notes](https://github.com/cloudflare/circl/releases)
- [Commits](https://github.com/cloudflare/circl/compare/v1.6.1...v1.6.3)

---
updated-dependencies:
- dependency-name: github.com/cloudflare/circl
  dependency-version: 1.6.3
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/coreos/ignition/v2 from 2.25.1 to 2.26.0

Bumps [github.com/coreos/ignition/v2](https://github.com/coreos/ignition) from 2.25.1 to 2.26.0.
- [Release notes](https://github.com/coreos/ignition/releases)
- [Changelog](https://github.com/coreos/ignition/blob/main/docs/release-notes.md)
- [Commits](https://github.com/coreos/ignition/compare/v2.25.1...v2.26.0)

---
updated-dependencies:
- dependency-name: github.com/coreos/ignition/v2
  dependency-version: 2.26.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yussufsh committed:
update the ignition version in docs

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

yussufsh committed:
Update ignition spec to 3.6.0

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

dependabot[bot] committed:
Bump goreleaser/goreleaser-action from 6 to 7

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 6 to 7.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v6...v7)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.38.2 to 2.39.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.38.2 to 2.39.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.38.2...v2.39.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-version: 2.39.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/kisielk/errcheck from 1.9.0 to 1.10.0

Bumps [github.com/kisielk/errcheck](https://github.com/kisielk/errcheck) from 1.9.0 to 1.10.0.
- [Release notes](https://github.com/kisielk/errcheck/releases)
- [Commits](https://github.com/kisielk/errcheck/compare/v1.9.0...v1.10.0)

---
updated-dependencies:
- dependency-name: github.com/kisielk/errcheck
  dependency-version: 1.10.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.39.0 to 2.40.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.39.0 to 2.40.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.39.0...v2.40.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-version: 2.40.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump google.golang.org/grpc in the go_modules group across 1 directory

Bumps the go_modules group with 1 update in the / directory: [google.golang.org/grpc](https://github.com/grpc/grpc-go).


Updates `google.golang.org/grpc` from 1.79.2 to 1.79.3
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.79.2...v1.79.3)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-version: 1.79.3
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump crazy-max/ghaction-import-gpg from 6 to 7

Bumps [crazy-max/ghaction-import-gpg](https://github.com/crazy-max/ghaction-import-gpg) from 6 to 7.
- [Release notes](https://github.com/crazy-max/ghaction-import-gpg/releases)
- [Commits](https://github.com/crazy-max/ghaction-import-gpg/compare/v6...v7)

---
updated-dependencies:
- dependency-name: crazy-max/ghaction-import-gpg
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
